### PR TITLE
Ensure resourceconstructor test includes beans.xml

### PR DESCRIPTION
**Pull request by andymc12**
This change adds the beans.xml file to the deployed WAR for the `resourceconstructor` test.  The beans.xml was added as part of challenge issue #938, but from my testing it was not added to the WAR file.  This change fixes that.

Since this only fixes a bug in a TCK test and makes no changes to spec or API, I'd like to request fast-track.

### CODE REVIEW

**Review by mkarg — COMMENTED** [review]
I vote +0 here as I understand your fix is beneficial but I did not yet dive into the TCK source deeply, so I cannot judge the complete correctness.

**Review by jansupol — APPROVED** [review]

File: jaxrs-tck/src/main/java/jakarta/ws/rs/tck/common/JAXRSCommonClient.java [hunk 1]
```diff
@@ -533,15 +533,19 @@ protected static String buildHeaderMediaType(String header, MediaType type) {
...
 
-  public static String editWebXmlString(InputStream inStream) throws IOException{
+  public static String toString(InputStream inStream) throws IOException{
     String line;
-    String webXmlTemplate = "";
+    String str = "";
     try (BufferedReader bufReader = new BufferedReader(new InputStreamReader(inStream, StandardCharsets.UTF_8))) {
       while ((line = bufReader.readLine()) != null) {
-        webXmlTemplate = webXmlTemplate + line + System.lineSeparator();
+        str = str + line + System.lineSeparator();
```
Comment (spericas) [inline jaxrs-tck/src/main/java/jakarta/ws/rs/tck/common/JAXRSCommonClient.java:541]: I know this is just a TCK, but should we replace this by a `StringBuilder`?

Comment (andymc12) [reply jaxrs-tck/src/main/java/jakarta/ws/rs/tck/common/JAXRSCommonClient.java:541]: Sure - I think I just left the existing logic alone there, but we could optimize, modernize, and economize it to be:
```
  public static String toString(InputStream inStream) throws IOException{
    try (BufferedReader bufReader = new BufferedReader(new InputStreamReader(inStream, StandardCharsets.UTF_8))) {
      return bufReader.lines().collect(Collectors.joining(System.lineSeparator()));
    }
  }

  public static String editWebXmlString(InputStream inStream) throws IOException{
    return toString(inStream).replaceAll("servlet_adaptor", servletAdaptor);
  }
```

**Review by spericas — APPROVED** [review]

**Comment by andymc12** [comment]
@jansupol could you take another look when you have a minute or two?  The only change since you last approved was to modify the string concatenation logic to use `Collectors.joining(...)` instead.  Thanks!

**Review by jansupol — APPROVED** [review]

**Pull request merged**